### PR TITLE
feat: allow customizing stryker directory and branch name

### LINKED ISSUE #1: Add option to specify stryker conf location
Actually, the script gets `stryker.conf.js` from the root directory location.

It would be nice to be able to provide the path to that file in case it is not in the root directory.

### LINKED ISSUE #2: Add option to specify default branch
Actually, the script compare current branch with "master" one.

It could be nice to be able to provide the targeted default branch in case "master" is not  the default one.

### PR DESCRIPTION
**Pull request by PauloGoncalvesBH**
'--path' and '--branch' commands will allow multiple projects to use the library.

@tverhoken Due to my weak knowledge in .ts I was unable to create the tests, I'm sorry for that.
My English is weak, if you believe that you need to adjust some syntax, please inform me about it.

This PR will be very good for my project (https://github.com/PauloGoncalvesBH/ServeRest) and it is what is missing for me to use this great library.

fix #1
fix #2

### CODE REVIEW

**Comment by PauloGoncalvesBH** [comment]
@tverhoken if you don't have a lot of time, what do you think about adding me as a collaborator? I can help maintain the stryker-diff-runner project and build pipelines for continuous delivery.

**Comment by tverhoken** [comment]
Hello @PauloGoncalvesBH,

Thanks a lot for contributing to this repository. I must confess that I did not take care enough about this project's update (@stryker-mutator/core is up to version 4.0.0 and here is still stuck to 2.0.1 😬 ).

No problem about the `.ts` syntax and tests. If it's ok for you I can suggest an example or we could try some pair programming to do it in order to satisfy minimum project quality.



> @tverhoken if you don't have a lot of time, what do you think about adding me as a collaborator? I can help maintain the stryker-diff-runner project and build pipelines for continuous delivery.

Why not. Let's talk about it once that PR is done 😉

File: src/index.ts [hunk 1]
```diff
@@ -4,20 +4,32 @@ import { exec } from "child_process";
 import { join } from "path";
 
 export default function run(commandArgs: string[]) {
-  const strykerConfPath = join(process.cwd(), "stryker.conf.js");
+
+  const args = commandArgs.slice(2);
+  let branch;
+  let path;
+
+  mapAllArgumentsToAnArray(args).forEach((element) => {
```
Comment (tverhoken) [inline src/index.ts:12]: Maybe we could avoid looping over the entire array and use `.filter` or `.find` for both seeking arguments ?

File: src/index.ts [hunk 2]
```diff
@@ -4,20 +4,32 @@ import { exec } from "child_process";
...
+    switch (element[0]) {
+      case "branch":
+        branch = element[1];
+        break;
+      case "path":
+        path = element[1];
+    }
+  });
```
Comment (tverhoken) [inline src/index.ts:20]: Something like :

```suggestion
branch = mapAllArgumentsToAnArray(args).find(argArray => argArray[0] === "branch");
branch = mapAllArgumentsToAnArray(args).find(argArray => argArray[0] === "path");
```

What do you think ?

We could also declare `branch` and `path` as `const`.

File: src/index.ts [hunk 3]
```diff
@@ -4,20 +4,32 @@ import { exec } from "child_process";
 import { join } from "path";
 
 export default function run(commandArgs: string[]) {
```
Comment (tverhoken) [inline src/index.ts:6]: For unit tests, we should add 2 tests that assert the default branch use for `exec` first arg and the provided argument when targeting another branch.

Same for the Stryker's configuration path.

Are you familiar with unit testing and mocks ? Or would you like an example with one of the test case involved ?

**Review by tverhoken — CHANGES_REQUESTED** [review]
Nice job here 😃 

A little refactoring maybe but it will be safer to have unit tests before doing so. I'm a little strict about project quality and tests so I can't let new behavior uncovered. I'm ok to help with it if you are not familiar about unit test practices.

**Review by tverhoken — APPROVED** [review]

**Comment by tverhoken** [comment]
Hi @PauloGoncalvesBH 

I allowed myself to push to your repo in order to make this PR go.

Feel free to look at commits I did and to contact me if you wanna exchange about it.

**Pull request merged**